### PR TITLE
Mobile NeRF general fixes

### PR DESCRIPTION
**Pull request by RodrigoHolztrattner-QuIC**
## Description

- Fix combo mode not properly using the right MLP weights.
- Fix 'image_layout_transition()' at 'setup_attachment()' throwing an error when using _VK_IMAGE_LAYOUT_GENERAL_.
- Initialize _VkAttachmentDescriptions_ to fix validation issues.
- Use _vkb::sg::Image::Other_ instead of _vkb::sg::Image::Color_ so textures are properly loaded as linear color space instead sRGB.
- Other clean up and modification:
    1. Remove `weights_buffer_ref` & `uniform_buffer_ref`, which are raw ptrs pointing to unique ptrs in vectors. This is unsafe, as everything will mess up when vector gets reallocated. Use indices as references instead.
    2. No need to update weights buffers for every frame, only upon their creation.
    3. No need to have samplers for G-buffer attachments because we directly use subpass load, removed.
    4. No need for mlp.comp shader, removed.
    5. No need for `aligned_size()` function, removed.
    6. No need to include "stb_image.h" in mobile_nerf.cpp.
    7. Remove debug code in shaders.

## General Checklist:

Please ensure the following points are checked:

- [x] My code follows the [coding style](https://github.com/KhronosGroup/Vulkan-Samples/tree/main/CONTRIBUTING.adoc#Code-Style)
- [x] I have reviewed file [licenses](https://github.com/KhronosGroup/Vulkan-Samples/tree/main/CONTRIBUTING.adoc#Copyright-Notice-and-License-Template)
- [x] I have commented any added functions (in line with Doxygen)
- [x] I have commented any code that could be hard to understand
- [x] My changes do not add any new compiler warnings
- [x] My changes do not add any new validation layer errors or warnings
- [x] I have used existing framework/helper functions where possible
- [x] My changes do not add any regressions
- [x] I have tested every sample to ensure everything runs correctly
- [x] This PR describes the scope and expected impact of the changes I am making

 Note: The Samples CI runs a number of checks including:
 - [x] I have updated the header Copyright to reflect the current year (CI build will fail if Copyright is out of date)
 - [x] My changes build on Windows, Linux, macOS and Android. Otherwise I have [documented any exceptions](https://github.com/KhronosGroup/Vulkan-Samples/tree/main/CONTRIBUTING.adoc#General-Requirements)

 If this PR contains framework changes:
 - [ ] I did a full batch run using the `batch` command line argument to make sure all samples still work properly

## Sample Checklist

If your PR contains a new or modified sample, these further checks must be carried out *in addition* to the General Checklist:
- [x] I have tested the sample on at least one compliant Vulkan implementation
- [ ] If the sample is vendor-specific, I have [tagged it appropriately](https://github.com/KhronosGroup/Vulkan-Samples/tree/main/CONTRIBUTING.adoc#General-Requirements)
- [ ] I have stated on what implementation the sample has been tested so that others can test on different implementations and platforms
- [x] Any dependent assets have been merged and published in downstream modules
- [ ] For new samples, I have added a paragraph with a summary to the appropriate chapter in the readme of the folder that the sample belongs to [e.g. api samples readme](https://github.com/KhronosGroup/Vulkan-Samples/blob/main/samples/api/README.adoc)
- [ ] For new samples, I have added a tutorial README.md file to guide users through what they need to know to implement code using this feature. For example, see [conditional_rendering](https://github.com/KhronosGroup/Vulkan-Samples/tree/main/samples/extensions/conditional_rendering)
- [ ] For new samples, I have added a link to the [Antora navigation](https://github.com/KhronosGroup/Vulkan-Samples/blob/main/antora/modules/ROOT/nav.adoc) so that the sample will be listed at the Vulkan documentation site

### CODE REVIEW

File: samples/general/mobile_nerf/mobile_nerf.cpp [hunk 1]
```diff
@@ -623,7 +614,7 @@ void MobileNerf::setup_nerf_framebuffer_baseline()
 	framebuffer_create_info.sType                   = VK_STRUCTURE_TYPE_FRAMEBUFFER_CREATE_INFO;
 	framebuffer_create_info.pNext                   = NULL;
 	framebuffer_create_info.renderPass              = render_pass_nerf;
-	framebuffer_create_info.attachmentCount         = static_cast<uint32_t>(views.size());
+	framebuffer_create_info.attachmentCount         = views.size();
```
Comment (asuessenbach) [inline samples/general/mobile_nerf/mobile_nerf.cpp]: Why do you remove the `static_cast`?

File: samples/general/mobile_nerf/mobile_nerf.cpp [hunk 2]
```diff
@@ -729,7 +758,7 @@ void MobileNerf::build_command_buffers_baseline()
 	render_pass_begin_info.renderArea.offset.y      = 0;
 	render_pass_begin_info.renderArea.extent.width  = width;
 	render_pass_begin_info.renderArea.extent.height = height;
-	render_pass_begin_info.clearValueCount          = static_cast<uint32_t>(clear_values.size());
+	render_pass_begin_info.clearValueCount          = clear_values.size();
```
Comment (asuessenbach) [inline samples/general/mobile_nerf/mobile_nerf.cpp]: Again, why do you remove the `static_cast`?

File: samples/general/mobile_nerf/mobile_nerf.cpp [hunk 3]
```diff
@@ -769,13 +798,24 @@ void MobileNerf::build_command_buffers_baseline()
 			vkCmdBindPipeline(draw_cmd_buffers[i], VK_PIPELINE_BIND_POINT_GRAPHICS, model.pipeline_first_pass);
 			// If deferred, only use the first descriptor bounded with the model
 			// If forward, each model has the swapchan number of descriptor
-			int descriptorIndex = use_deferred ? 0 : static_cast<uint32_t>(i);
+			int descriptorIndex = use_deferred ? 0 : i;
```
Comment (asuessenbach) [inline samples/general/mobile_nerf/mobile_nerf.cpp]: As above.

File: shaders/mobile_nerf/mlp_combo.frag [hunk 4]
```diff
@@ -0,0 +1,238 @@
...
+
+        int bias_0_ind = WEIGHTS_0_COUNT + WEIGHTS_1_COUNT + WEIGHTS_2_COUNT;
+        vec4 intermediate_one[4] = vec4[](
+           weights_arr[nonuniformEXT(idx)].data[bias_0_ind/4],
+           weights_arr[nonuniformEXT(idx)].data[bias_0_ind/4 + 1],
+           weights_arr[nonuniformEXT(idx)].data[bias_0_ind/4 + 2],
+           weights_arr[nonuniformEXT(idx)].data[bias_0_ind/4 + 3]
+        );
+
+        #if 0
```
Comment (asuessenbach) [inline shaders/mobile_nerf/mlp_combo.frag]: Is this #if 0-ed section worth to be kept? Why?

File: shaders/mobile_nerf/mlp_combo.frag [hunk 5]
```diff
@@ -0,0 +1,238 @@
...
+        int bias_1_ind = WEIGHTS_0_COUNT + WEIGHTS_1_COUNT + WEIGHTS_2_COUNT +
+                         BIAS_0_COUNT;
+        vec4 intermediate_two[4] = vec4[](
+           weights_arr[nonuniformEXT(idx)].data[bias_1_ind/4],
+           weights_arr[nonuniformEXT(idx)].data[bias_1_ind/4 + 1],
+           weights_arr[nonuniformEXT(idx)].data[bias_1_ind/4 + 2],
+           weights_arr[nonuniformEXT(idx)].data[bias_1_ind/4 + 3]
+        );
+
+         #if 0
```
Comment (asuessenbach) [inline shaders/mobile_nerf/mlp_combo.frag]: As above.

File: samples/general/mobile_nerf/mobile_nerf.cpp (hunk 1 above)
Comment (RodrigoHolztrattner-QuIC) [reply samples/general/mobile_nerf/mobile_nerf.cpp]: These got lost internally, added back, thanks for noticing.

File: samples/general/mobile_nerf/mobile_nerf.cpp (hunk 2 above)
Comment (RodrigoHolztrattner-QuIC) [reply samples/general/mobile_nerf/mobile_nerf.cpp]: These got lost internally, added back, thanks for noticing.

File: samples/general/mobile_nerf/mobile_nerf.cpp (hunk 3 above)
Comment (RodrigoHolztrattner-QuIC) [reply samples/general/mobile_nerf/mobile_nerf.cpp]: These got lost internally, added back, thanks for noticing.

File: shaders/mobile_nerf/mlp_combo.frag (hunk 5 above)
Comment (RodrigoHolztrattner-QuIC) [reply shaders/mobile_nerf/mlp_combo.frag]: Removed these and all other occurrences.

File: shaders/mobile_nerf/mlp_combo.frag (hunk 4 above)
Comment (RodrigoHolztrattner-QuIC) [reply shaders/mobile_nerf/mlp_combo.frag]: Removed these and all other occurences

**Review by asuessenbach — APPROVED** [review]

**Review by SaschaWillems — APPROVED** [review]
Code changes look fine to me. Works on my RTX 4070 on WIn11.

**Pull request merged**